src/commander/commander.ts

```diff
@@ -0,0 +1,253 @@
+        `Unknown command "${unknownCommand || ""}"; Available ones are ${Object.keys(commandConfigs).join(
+          ", ",
+        )}. ${helpStr}`,
+      );
+    }
+  });
+
+  root.hook("postAction", (thisCommand, actionCommand) => {
+    let currentCommand = actionCommand;
+    const commandOptions = actionCommand.optsWithGlobals();
+    const commandStack = []; // This'll contain command and (optionally) preset, e.g. ["bench", "polkadot"]
+    while (currentCommand.parent) {
+      commandStack.unshift(currentCommand.name());
+      currentCommand = currentCommand.parent;
+    }
+    const commandOptionsRaw = optionValuesToFlags(commandOptions);
+
+    root.parseResults = {
+      command: commandStack[0],
+      commandOptions,
+      commandOptionsRaw,
+      commandArgs: actionCommand.args,
+      ignoredCommand: botPullRequestIgnoreCommands.includes(commandStack[0]),
+      unknownCommand,
+      parsedCommand,
+    };
+  });
+
+  return root;
+}
+
+function addPresetOptions(cfg: {
+  presetCommand: Command;
+  presetConfig: NonNullable<CmdJson["command"]["presets"]>[keyof NonNullable<CmdJson["command"]["presets"]>];
+  commandKey: string;
+  commandConfig: CmdJson;
+  actionCallBack: (genericCommand: GenericCommand) => void;
+}): Command {
+  const { presetConfig, presetCommand, commandConfig, commandKey, actionCallBack } = cfg;
+  if (presetConfig.description) presetCommand.description(presetConfig.description);
+
+  if (presetConfig.args) {
+    for (const [argKey, argConfig] of Object.entries(presetConfig.args)) {
+      const option = convertOption(argKey, argConfig);
+      presetCommand.addOption(option);
+    }
+  }
+
+  presetCommand
+    .exitOverride((e) => {
+      throw new Error((e as CommanderError).message.replace("error: ", ""));
+    })
+    .action((commandOptions: OptionValues, cmd: Command) => {
+      // extract global variable from the rest of the default options
+      const { variable: variables, ...rest } = cmd.optsWithGlobals() as { [key: string]: Record<string, string> };
+      actionCallBack(getGenericCommand({ commandKey, commandConfig, commandOptions: rest, variables }));
+    });
+
+  return presetCommand;
+}
+
+function convertOption(
+  argKey: string,
+  argConfig: {
+    [k: string]: unknown;
+  },
+): Option {
+  const flags = `--${argKey} <value>`;
+
+  // argConfig.label should always be string, but TS doesn't believe it
+  const option = new Option(flags, argConfig.label as string);
+
+  if (Array.isArray(argConfig.type_one_of)) {
+    option.choices(argConfig.type_one_of);
+    option.default(argConfig.type_one_of[0]);
+  } else if (typeof argConfig.type_string === "string" && argConfig.type_string) {
+    option.default(argConfig.type_string);
+  } else if (typeof argConfig.type_rule === "string") {
+    // FIXME: currently rules are defined as "/^([a-z_]+)([:]{2}[a-z_]+)?$/", with extra slashes
+    const ruleRegex = new RegExp(argConfig.type_rule.substring(1, argConfig.type_rule.length - 1));
```

Comment:
Are we planning to merge this and https://github.com/paritytech/command-bot-scripts/pull/23 separately? Otherwise, what stopping us from changing the rules now?

Reply:
ugh, i didn't notice that FIXME. Ya totally better to fix it in this PR
And yes, this goes PR first (as it has hardcoded `monorepo` branch). Then merging `monorepo` to `main` of scripts. Then changing here `monorepo` back to `main`